jina/parser.py

```diff
@@ -517,9 +517,6 @@ def set_client_cli_parser(parser=None):
     gp1.add_argument('--mode', choices=list(ClientMode), type=ClientMode.from_string,
                      # required=True,
                      help='the mode of the client and the server')
-    gp1.add_argument('--top-k', type=int,
```

Comment:
why not remove `default=10` instead of removing the whole arg?

Reply:
Why do not we do this? 

we can keep that top_k  in the client (with a default to None) and move the change in the RESTGatewayPea to the search method in request.py

Reply:
As per our current changes, `top_k` parameter was not used and so we removed it

Reply:
@bhavsarpratik add the parameter back with no default